### ISSUE TITLE
Need to verify another 2015 EITC amount

### ISSUE THREAD

**Issue by martinholmer** [open-source-economics/Tax-Calculator#572]
We added 2015 to the Internet-TAXSIM versus Tax-Calculator validation test suite in pull request #561.

There were no FICA differences between the two models using the c15.in sample of 100,000 randomly generated filing units.  However, there were 617 filing units with income tax differences of more than one cent.  The largest income tax difference was $77.93 for the unit with id 74480 and all that difference was caused by a difference in the two model's estimates of the EITC amount.

We need to compute this unit's EITC amount using the 2015 forms to verify that the Tax-Calculator current-law-policy parameters and logic are correct.

If we can confirm that the Tax-Calculator is computing the correct EITC for this filing unit, then we need to send a scanned image of the filled-out forms to Dan Feenberg at NBER.  That is the Internet-TAXSIM bug-reporting procedure.  Also, send Dan the 22-variable input line for the filing unit with id 74480, and remind him that we are using Internet-TAXSIM option ```56 1``` to generate results that we compare with Tax-Calculator results.

Sean, can you undertake this validation task?

cc @MattHJensen @feenberg @Amy-Xu @GoFroggyRun

**Comment by GoFroggyRun** [comment]
@martinholmer Thanks for pointing out this issue. I might not be able to start looking in to this immediately, but please expect this issue to be verified no latter than this Thursday.

**Comment by martinholmer** [comment]
Sean said:
> Thanks for pointing out this issue. I might not be able to start looking in to this immediately, but please expect the issue to be verified no later than this Thursday.

@GoFroggyRun,  That sounds great.  Thanks for looking into this matter.

**Comment by GoFroggyRun** [comment]
@martinholmer:

I just checked record with id 74480. The unit has earned income $26,700, which yields a credit of $4,896; while his AGI is of $49,270, which yields a credit of $147. The smaller of which will be the EITC of that person, which is $147. Our calculator gives the EITC output of $148.22, while Internet Taxsim indicates $70.29 to be the EITC amount. 

cc @MattHJensen @feenberg

**Comment by martinholmer** [comment]
Sean said:
> I just checked record with id 74480. The unit has earned income $26,700, which yields a credit of $4,896; while his AGI is of $49,270, which yields a credit of $147. The smaller of which will be the EITC of that person, which is $147. Our calculator gives the EITC output of $148.22, while Internet-TAXSIM indicates $70.29 to be the EITC amount.

Sean,  Thanks for the quick verification of this case.

The Tax-Calculator amount of $148.22, which is computed using a formula, is very close to the IRS table-lookup value of $147, certainly much closer than the Internet-TAXSIM amount of $70.29.  So, it would seem that the 2015 EITC policy parameters may be a little off in Internet-TAXSIM.  If others agree with this assessment, can you send the filled-out tax forms (and the input record and both the output records) to Dan at NBER?

cc @GoFroggyRun @feenberg @MattHJensen @Amy-Xu

**Comment by GoFroggyRun** [comment]
@martinholmer 
Sure.

@feenberg 
Please expect an email with rotated forms tomorrow.

**Comment by feenberg** [comment]
On Wed, 3 Feb 2016, Martin Holmer wrote:

> 
> Sean said:
>
>       I just checked record with id 74480. The unit has earned income $26,700,
>       which yields a credit of $4,896; while his AGI is of $49,270, which yields
>       a credit of $147. The smaller of which will be the EITC of that person,
>       which is $147. Our calculator gives the EITC output of $148.22, while
>       Internet-TAXSIM indicates $70.29 to be the EITC amount.
> 
> Sean, Thanks for the quick verification of this case.
> 
> The Tax-Calculator amount of $148.22, which is computed using a formula, is very close
> to the IRS table-lookup value of $147, certainly much closer than the Internet-TAXSIM
> amount of $70.29. So, it would seem that the 2015 EITC policy parameters may be a
> little off in Internet-TAXSIM. If others agree with this assessment, can you send the
> filled-out tax forms (and the input record and both the output records) to Dan at
> NBER?

The latest version of the SAS calculator has revised parameters - they 
were released by the IRS several weeks ago.

I'll upload the new version as soon as I can make it and the fortran 
calculator match - something that is much harder than I thought. The 
treatment of taxpayers with no taxable income, but with itemized 
deductions is currently vexing me.

dan

**Comment by martinholmer** [comment]
Dan posted the newest version of Internet TAXSIM before noon on 17-Feb-2016.  Using this newest version, I redid the cross-model validation test using the c15.in sample of 100,000 randomly-generated filing units with tax year 2015.

The largest income tax difference is now 4.21 dollars and the largest EITC difference is also 4.21. There are now 604 differences each in the income tax and EITC amounts. The number of differences is down from 617 and the largest difference is down from 77.93 dollars.

So, this is progress in the cross-model validation work for the c15.in sample, but it would appear that there are still (small) differences in the logic of the two models. This seems to be the case because when using the c13.in and c14.in samples (for 2013 and 2014, respectively) we found no differences of more than one cent in either the EITC amount or the total income tax liability.

But it is not clear which model needs minor adjustment.  In order to identify which model is slightly off, it would seem that we need to verify a filing unit by filling out 2015 tax forms in order to see which model needs adjusting (or maybe both need adjusting).

Sean, I'm hoping you can do this again just like you did earlier in this issue.  Here is the filing unit (in Internet-TAXSIM input format) that needs to be verified:
```
$ awk '$1==36252' c15.in
36252 2015 0 3 3 0 18400 0 3000 0 0 0 0 0 1900 100 1000 1000 3 5000 -3000 -4000
```

And here are the latest results from Internet TAXSIM and Tax Calculator (via simtax.py):
```
$ awk '$1==36252' c15.in.out-taxsim
36252. 2015 0 -8284.54 .00 2815.20 6.06 .00 15.30 19400.00
1000.00 .00 9250.00 16000.00 .00 .00 .00 .00 .00 .00
.00 .00 2310.00 .00 5974.54 19400.00 .00 .00

$ awk '$1==36252' c15.in.out-simtax
36252. 2015 0 -8280.33 0.00 2815.20 6.06 0.00 15.30 19400.00
1000.00 0.00 0.00 16000.00 0.00 0.00 0.00 0.00 0.00 0.00
0.00 0.00 2310.00 0.00 5970.33 19400.00 0.00 0.00
```

cc @MattHJensen @feenberg @GoFroggyRun

**Comment by GoFroggyRun** [comment]
@martinholmer :

Before working on the forms, I think such information might be helpful:
>
I just checked record with id 74480. The unit has earned income $26,700, which yields a credit of $4,896; while his AGI is of $49,270, which yields a credit of $147. The smaller of which will be the EITC of that person, which is $147. Our calculator gives the EITC output of $148.22, while Internet Taxsim indicates $70.29 to be the EITC amount.

For that same record with id 74480, I just checked that the EITC amount in the latest TAXSIM output for that unit is $146.11, and the difference on EITC between two calculators for that record remain over 1 cent. It seems to me that, without filing a new unit, the record with id 74480 is still being helpful in this case. 
At least for this record, the difference here seems coming from how the two calculators are approximating the discrete tax table by using a continuous function. Any ideas how to deal with issues like this?

@martinholmer @MattHJensen @feenberg

**Comment by martinholmer** [comment]
After looking at the differences between the two models for filing unit 74480 (for which he filled out the tax forms earlier), where TAXSIM EITC is lower by $2.11, and hence, TAXSIM income tax is higher by $2.11, Sean said:
> At least for this record [id=74480], the difference here seems coming from how the two calculators are approximating the discrete tax table by using a continuous function. Any ideas how to deal with issues like this?

Well, Sean is correct in thinking that these differences **may** be caused the two models approximating the EITC lookup table in different ways.  But I would be very surprised if this turned out to be true.  The reason I would be surprised is that for both the 2013 sample c13.in and the 2014 sample c14.in there were no EITC differences of more that one cent.  So, how could the exact agreement between the two models in approximating the EITC lookup table for 2013 and 2014, disappear in 2015?  Seems unlikely, but I must admit I don't (and do not want) access to the Internet TAXSIM source code.

cc @MattHJensen @feenberg @GoFroggyRun

**Comment by feenberg** [comment]
On Wed, 17 Feb 2016, Sean.Wang wrote:

> 
> @martinholmer :
> 
> Before working on the forms, I think such information might be helpful:
>
>       I just checked record with id 74480. The unit has earned income $26,700,
>       which yields a credit of $4,896; while his AGI is of $49,270, which yields
>       a credit of $147. The smaller of which will be the EITC of that person,
>       which is $147. Our calculator gives the EITC output of $148.22, while
>       Internet Taxsim indicates $70.29 to be the EITC amount.
> 
> For that same record with id 74480, I just checked that the EITC amount in the latest
> TAXSIM output for that unit is $146.11, and the difference on EITC between two
> calculators for that record remain over 1 cent. It seems to me that, without filing a
> new unit, the record with id 74480 is still being helpful in this case.
> At least for this record, the difference here seems coming from how the two
> calculators are approximating the discrete tax table by using a continuous function.
> Any ideas how to deal with issues like this?

Taxsim does not use the tax tables, but uses the X,Y and Z table. I don't 
think we would consider changing that, since it makes marginal tax rates 
more difficult to calculate. Furthermore, I don't think it will improve 
the aggregate estimate. My logic is that the rounding adds a little noise 
to each record. Getting that noise right won't improve the aggregate by an 
interesting amount.

dan

**Comment by GoFroggyRun** [comment]
After looking at records with id 36252 and 20089, I found that:

1. For record with id 20089: two calculators agree on the AGI of amount $19,300, and thus this should yield EITC of amount $5,292 according to the EITC table. Our calculator approximates the amount to be $5,295.28, while TAXSIM yields an amount of $5,301.60.  

2. Similarly, for record with id 36252: two calculators agree on the AGI of amount $19,400, and thus this should yield EITC of amount $5,965 according to the EITC table. Our calculator approximates the amount to be $5,968.22, while TAXSIM yields an amount of $5,974.54.  

I'm somewhat confused by this. Since two calculators did not essentially change their ways of approximating EITC, why would the EITC amount doesn't agree in 2015 but in 2013, 2014. I speculate that the errors are still coming from parameters regarding EITC, possibly the EITC phaseout threshold.

@martinholmer @feenberg @MattHJensen 

Attached is input, output for Taxsim and output for our calculator respectively: 
```
20089 2015 0 3 2 0 15300 0 1000 0 0 0 0 0 2119 3935 11000 4000 2 2096 2000 -3000
20089. 2015 0 -7146.60 .00 2340.90 6.06 .00 15.30 19300.00 4000.00 .00 9250.00 12000.00 .00 .00 .00 .00 .00 .00 .00 .00 1845.00 .00 5301.60 19300.00 .00 .00 2340.90 20301.01 .00 .00 .00 .00 .00 .00 .00 .00 .00 .00 .00
20089. 2015 0 -7140.28 0.00 2340.90 6.06 0.00 15.30 19300.00 4000.00 0.00 0.00 12000.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 1845.00 0.00 5295.28 19300.00 0.00 0.00
```
```
36252 2015 0 3 3 0 18400 0 3000 0 0 0 0 0 1900 100 1000 1000 3 5000 -3000 -4000
36252. 2015 0 -8284.54 .00 2815.20 6.06 .00 15.30 19400.00 1000.00 .00 9250.00 16000.00 .00 .00 .00 .00 .00 .00 .00 .00 2310.00 .00 5974.54 19400.00 .00 .00 2815.20 22401.01 .00 .00 .00 .00 .00 .00 .00 .00 .00 .00 .00
36252. 2015 0 -8278.22 0.00 2815.20 6.06 0.00 15.30 19400.00 1000.00 0.00 0.00 16000.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 0.00 2310.00 0.00 5968.22 19400.00 0.00 0.00
```

**Comment by GoFroggyRun** [comment]
@martinholmer:
Thanks for the prompt updates in PR #605 . I have, however, a few concerns regarding this PR.

1. After updating the master, again, we look at record id = 74480: this unit now has EITC of amount $152.43, which is even further off from the actual look-up table value, $147. (Previously, we have $146.11/ $148.22, depending on tuning of the parameter `_EITC_ps`.)

2. I'm not quite comfortable with the modification of parameter `_EITC_ps_MarriedJ` in #605. If you look into previous tax forms, say [2014](https://www.irs.gov/pub/irs-prior/i1040gi--2014.pdf), and compare the previous values in the master and the actual worksheet, you would find that the amount of `_EITC_ps_MarriedJ` actually appears in worksheet B, instead of worksheet A (where I presume you obtained the updated array `5500, 5500, 5500, 5500` from).

3. For parameter `_EITC_ps`, although `8250, 18150, 18150, 18150` does correspond to the actual phaseout threshold appears in the EITC table, this is not the optimal way of setting the parameter. Since we are using continuous function to approximate the discrete table, we need to modify the actual threshold a little bit to obtain a better approximation. If you, again, take a look at the threshold for previous year, say 2014, `8150, 17850, 17850, 17850` (actual phaseout threshold) was modified and recored as `8110, 17830, 17830, 17830`  (modified phaseout threshold in master).

My current strategy is to modify `_EITC_ps` to obtain some optimized approximation, since this is the only parameter that seems to be "flexible". Before that, I guess we still need to figure out what the amount of `_EITC_ps_MarriedJ` ought to be.

**Comment by martinholmer** [comment]
Issue #572 has been resolved by the merging of pull requests #605 and #613 and #614 into the master branch.
